### ISSUE TITLE
Dependency Conflict with typing_extensions

### ISSUE THREAD

**Issue by junah201** [junah201/caliber#1]
When using vellox with the openai module, a module dependency conflict occurs.

openai requires typing-extensions>=4.11, <5
vellox requires typing_extensions>=4.7.1,<5

```bash
Installing collected packages: typing_extensions, vellox
Attempting uninstall: typing_extensions
Found existing installation: typing_extensions 4.12.2
Uninstalling typing_extensions-4.12.2:
Successfully uninstalled typing_extensions-4.12.2
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
fastapi 0.112.2 requires typing-extensions>=4.8.0, but you have typing-extensions 4.7.1 which is incompatible.
openai 1.42.0 requires typing-extensions<5,>=4.11, but you have typing-extensions 4.7.1 which is incompatible.
Successfully installed typing_extensions-4.7.1 vellox-0.1.2
```

> This issue was reported by Donghyun Lim. by using medium comment
![image](https://github.com/user-attachments/assets/b8faddfd-31bf-49f0-b7a2-ea29cbaacc16)

**Comment by junah201** [comment]
I have confirmed that the issue is reproducible

**Comment by junah201** [comment]
The typing-extensions dependency changed in this commit 2dfccb314c55e744184c8342b33791b4cf474cd4 is not deployed.

**Comment by junah201** [comment]
This issue was resolved in 0.1.3.
